### PR TITLE
 #171187744 Fix profile and single request bugs 

### PR DESCRIPTION
**Pull request by nziokaivy**
#### What does this PR do?
Fixes profile and single request bugs

#### Description of Task to be completed?

- Fix the upload image issue that removes data
- Allow for an optional update of the profile
- Allow once a request has been rejected or approved to immediately get updated

#### How should this be manually tested?

- Clone this repository using the command `git clone https://github.com/andela/team-odd-bn-frontend.git`
- Then navigate into the project `cd team-odd-bn-frontend`
- Switch to the fix profile bug branch using the command: git checkout bg-fix-profile-bug-171187744
- Run the command `npm i `to install all the packages
- Run `npm run dev` to run the server
- Then finally, navigate to the profile page and click on the settings icon 

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#171187744](https://www.pivotaltracker.com/story/show/171187744)
###### Profile updated by only editing the gender and the line manager 

image

### CODE REVIEW

File: src/components/Profile.js [hunk 1]
```diff
@@ -92,20 +87,17 @@ export class Profile extends Component {
       });
     }
     if (image) {
-      this.setState({ imageURL: image });
+      this.setState({ ...this.state, imageURL: image });
```
Comment (NiyongaboEric) [inline src/components/Profile.js]: +1 @nziokaivy! Profile state is persistent.

**Review by NiyongaboEric — APPROVED** [review]
I think `async` and `await` will help us to update trip types status without refreshing the page. Good job @nziokaivy

**Review by victkarangwa — APPROVED** [review]
Nice job @nziokaivy

**Review by hezronkimutai — APPROVED** [review]
Nice job

File: src/components/CreateProfile.js [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+        bio: '',
+        formErrors: {
+            address: '',
+            department: '',
+            bio: '',
+        },
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleChangeInput = this.handleChangeInput.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
```
Comment (kipropbrian) [inline src/components/CreateProfile.js]: No binding please

File: src/components/CreateProfile.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+            managerId: managerId || '',
+            address: address || '',
+            bio: bio || '',
+        });
+        }
+        if (image) {
+        this.setState({ ...this.state, imageURL: image });
+        }
+    }
+    handleChange(e) {
```
Comment (kipropbrian) [inline src/components/CreateProfile.js]: What change are you handling. Can you name your function with a better name.

**Review by william000000 — APPROVED** [review]
Nice Job @nziokaivy, Kindly work on the feedback Brian left, otherwise you are good to go

**Pull request merged**